File: sunday_night_ii/linked_list.py

class LinkedList:
    def __init__(self, x=None, next=None):
        self.val = x
        self.next = next

    def __str__(self):
        return "{}-->{}".format(self.val, self.next)


if __name__ == '__main__':
    my_ll = LinkedList(1)
    a = LinkedList(2)
    b = LinkedList(3)
    c = LinkedList(4)
    d = LinkedList(5)

    print(a.val)

    my_ll.next = a
    a.next = b
    b.next = c
    c.next = d

    print(my_ll)
    # print(type(my_ll))

File: sunday_night_ii/linked_list.java

public class linked_list {

    public static void main(String[] args) {
    Node my_ll = new Node(1, null);
    Node a = new Node(2, null);
    Node b = new Node(3, null);
    Node c = new Node(4, null);
    Node d = new Node(5, null);

    my_ll.next = a;
    a.next = b;
    b.next = c;
    c.next = d;

    System.out.println(my_ll.toString());

  }

}

class Node {
    int data;
    Node next;

    public Node(int data, Node next){
        this.data = data;
        this.next = next;
    }

    public String toString(){
      String result = data + "-->";
      if (next != null) {
          result += next.toString();
      } else if (next == null) {
        result += "null";
      }
      return result;
    }
}
